### PR TITLE
KAFKA-2813: selector doesn't close socket connection on non-IOExceptions

### PR DESCRIPTION
**Pull request by junrao**
Patched Selector.poll() to close the connection on any exception.

### CODE REVIEW

File: clients/src/main/java/org/apache/kafka/common/network/Selector.java [hunk 1]
```diff
@@ -284,16 +284,8 @@ public void poll(long timeout) throws IOException {
...
-                                addToStagedReceives(channel, networkReceive);
-                            }
-                        } catch (InvalidReceiveException e) {
-                            log.error("Invalid data received from " + channel.id() + " closing connection", e);
-                            close(channel);
-                            this.disconnected.add(channel.id());
-                            throw e;
-                        }
+                        while ((networkReceive = channel.read()) != null)
+                            addToStagedReceives(channel, networkReceive);
```
Comment (guozhangwang) [inline clients/src/main/java/org/apache/kafka/common/network/Selector.java:287]: Should we remove the `@throws InvalidReceiveException` in the comments of poll() then?

**Comment by guozhangwang** [comment]
Will this change will make another caller of Selector, Sender.run(), to not throw Exceptions any more? (line 130)

**Comment by gwenshap** [comment]
LGTM. @junrao, you can commit after system tests pass on this branch :+1:

File: clients/src/main/java/org/apache/kafka/common/network/Selector.java (hunk 1 above)
Comment (junrao) [inline clients/src/main/java/org/apache/kafka/common/network/Selector.java:287]: fixed

**Comment by junrao** [comment]
Sender may not get exception from Selector, but can get it from other places. However, it seems that we need to clean up Selectable since it never throws IOException in poll(). Will file a followup jira for that.

**Comment by guozhangwang** [comment]
LGTM.

**Comment by qa-prog** [comment]
See the official documentation for details: [Kafka Config Guide](https://kafka.apache.org/documentation/#configuration)

**Pull request closed without merging**